src/otx/core/model/entity/instance_segmentation.py

```diff
@@ -127,22 +134,101 @@ def _export_parameters(self) -> dict[str, Any]:
 class ExplainableOTXInstanceSegModel(OTXInstanceSegModel):
     """OTX Instance Segmentation model which can attach a XAI hook."""
 
-    def register_explain_hook(self) -> None:
-        """Register explain hook at the model backbone output."""
+    def forward_explain(
+        self,
+        inputs: T_OTXBatchDataEntity,
+    ) -> T_OTXBatchPredEntity | OTXBatchLossEntity:
+        """Model forward function."""
+        self.model.explain_fn = self.get_explain_fn()
+
+        # If customize_inputs is overridden
+        if self._customize_inputs != ExplainableOTXInstanceSegModel._customize_inputs:
+            customized_inputs = self._customize_inputs(inputs)
+        else:
+            customized_inputs = {inputs}
+        outputs = self._forward_explain_inst_seg(self.model, **customized_inputs)
+
+        return (
+            self._customize_outputs(outputs, inputs)
+            if self._customize_outputs != ExplainableOTXInstanceSegModel._customize_outputs
+            else outputs
+        )
+
+    @staticmethod
+    def _forward_explain_inst_seg(self, inputs: torch.Tensor, data_samples: SampleList, mode: str) -> dict:
```

Comment:
Please specify type of `self`

Reply:
Will do together with fixing precommit, thank you :)